batch/src/main/java/org/openmrs/analytics/FhirEtl.java

```diff
@@ -118,8 +118,16 @@
 	}
 	
 	static FhirSearchUtil createFhirSearchUtil(FhirEtlOptions options, FhirContext fhirContext) {
-		return new FhirSearchUtil(createOpenmrsUtil(options.getServerUrl() + options.getServerFhirEndpoint(),
-		    options.getUsername(), options.getPassword(), fhirContext));
+		return new FhirSearchUtil(createFhirStoreUtil(options.getFhirStoreUrl(), fhirContext),
+		        createOpenmrsUtil(options.getServerUrl() + options.getServerFhirEndpoint(), options.getUsername(),
+		            options.getPassword(), fhirContext));
+	}
+	
+	static FhirStoreUtil createFhirStoreUtil(String fhirStoreUrl, FhirContext fhirContext) {
+		if (GcpStoreUtil.matchesGcpPattern(fhirStoreUrl))
+			return new GcpStoreUtil(fhirStoreUrl, fhirContext);
+		else
+			return new FhirStoreUtil(fhirStoreUrl, fhirContext);
```

Comment:
The constructors here are defined to take an `IGenericClient` rather than `FhirContext`, so this should likely be something like:

```java
return new FhirStoreUtil(fhirStoreUrl, fhirContext.newRestfulGenericClient());
```

Reply:
Similarly, I tend to think that `if ... else` should always be blocks:

```java
if (condition) {

} else {

}
```

This makes it easier if additional actions need to be taken.

Reply:
> The constructors here are defined to take an `IGenericClient` rather than `FhirContext`, so this should likely be something like:
> 
> ```java
> return new FhirStoreUtil(fhirStoreUrl, fhirContext.newRestfulGenericClient());
> ```

Yeah, I accidentally committed in the middle of refactoring -_- thanks!